### ISSUE TITLE
Support soft delete

### ISSUE THREAD

**Issue by callmekatootie** [topcoder-platform/lookups-api#13]
Reference - https://github.com/topcoder-platform/admin-app/issues/130#issuecomment-593484409

We need to support soft deletes.

- Keep current deletion api as soft delete
- Let user pass `?destroy=true` through query parameters to actually delete the record

**Comment by callmekatootie** [comment]
@rootelement GET records is right now publicly available - no access needed.

Should the endpoint return records that have been soft deleted? Should we have a mechanism for admin users to fetch records that were soft deleted...?

**Comment by rootelement** [comment]
Sorry, I didn't see this comment. Can you easily differentiate between un-authed and authed GET? Can we have a filter on get to include deleted ones but have that only available to an authed GET?

**Comment by callmekatootie** [comment]
Let me know if I got this right:

- If there is no auth for the GET request, do not return records that have been soft deleted
- If there is auth for the GET request, we will also return records that have been soft deleted

**Comment by rootelement** [comment]
Yes, I believe that's true. Only an admin should be able to GET soft-deleted ones, and they should be hidden by default. There should be a query variable to include them somehow. Obviously only an admin should be able to soft delete or destroy.

**Comment by callmekatootie** [comment]
> There should be a query variable to include them somehow

So, if there's auth AND a query variable to get the soft deletes, only then we return them. If there's auth but the query param is not set, we don't need to return them. Got it :+1:

**Comment by callmekatootie** [comment]
Assigned to @meshde

**Comment by callmekatootie** [comment]
@rootelement The soft delete feature is now in `develop` branch. Please let me know when I can push it to `master`
